docs/general/accounts.md

```diff
@@ -1,170 +1,114 @@
-private key and a wallet (EOAs) and smart contracts, due to native Account
-Abstraction.
-
-Kakarot leverages this abstraction to provide a single account type that can be
-used for both EOA and CA use cases. This account type is managed by a single
-Starknet contract class (`account_contract.cairo`). This contract class defines
-the `__validate__` and `__execute__` entrypoints, which are used to send
-transactions from a wallet to Kakarot in a way that is compatible with the
-Starknet infrastructure. If you own the keys to an Ethereum address, you can use
-the `deploy_externally_owned_account` entrypoint to deploy a corresponding
-Starknet account contract, whose address is deterministic based on the Ethereum
-address.
-
-Although our backend makes no distinction between EOA and CA, the validation
-logic will reject any transaction that originates from a contract that has EVM
-code, as per the EVM specification. In the (extremely unlikely) event that a
-user sends a transaction from a contract account after mining the private key
-corresponding to an EOA, the transaction will be rejected.
-
-## Account storage
+# Accounts in Kakarot
+
+## Overview
+
+Kakarot leverages Starknet's native Account Abstraction to provide a single
+account type for both EOA (Externally Owned Account) and CA (Contract Account)
+use cases. This unified approach differs from the EVM, which distinguishes
+between EOAs and CAs.
+
+## Account Structure
+
+### Account Contract
+
+All accounts in Kakarot are managed by a single Starknet contract class
+(`account_contract.cairo`). This contract implements the required Starknet
+Account Abstraction entrypoints, but they're disabled in favor of the
+`execute_from_outside` entrypoint. `execute_from_outside` accepts transactions
+sent by relayers to leverage applicative paymaster features.
+
+### Account Storage
 
 Account contracts store the following information:
 
-- `storage`: A mapping from 32-byte keys to 32-byte values. This is used to
-  store the values stored in the EVM account storage.
-- `nonce`: A 64-bits value representing the number of transactions sent from the
-  account. Even though EOAs do not use the stored nonce for validation, it is
-  still stored in the account contract to ensure the stored nonce always matches
-  the protocol nonce, which is incremented by one by the sequencer for each
-  transaction.
-- `bytecode`: The EVM bytecode of the account. This is only used for CAs, as
-  EOAs do not have bytecode.
-- `version`: A 9-digit integer representing the version of the account contract,
-  in the format `major.minor.patch` (3 digits each).
-- `implementation`: The class hash of the current account contract class.
+- `storage`: A mapping from 32-byte keys to 32-byte values (EVM account
+  storage).
+- `nonce`: A 64-bit value representing the number of transactions sent from the
+  account.
+- `bytecode`: The EVM bytecode of the account (used only for CAs).
```

Comment:
"only used for", meaning empty, but cf previous comment, I'm not sure it's meaningful to make disctinction here about the structure of account. In EVM it's

```
balance: int
code: bytes
nonce: int
storage: Dict[int, int] = {}
```

and so in our account contracts